File: cse107/EIUFING.java
import java.util.Scanner;

public class EIUFING {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int randomNumber = scanner.nextInt();
        findFinger(randomNumber);
    }

    static void findFinger(int randomNumber) {
        String[] finger = { "cai", "tro", "giua", "ap ut", "ut", "ut", "ap ut", "giua", "tro", "cai" };
        String[] ind = { "trai", "phai" };

        int fingerNumber = randomNumber % 18;

        if (10 < fingerNumber && fingerNumber <= 18) {
            fingerNumber = 20 - fingerNumber;
        }
        int index = (fingerNumber > 5) ? 1 : 0;

        System.out.printf("Ngon %s cua ban tay %s", finger[fingerNumber - 1], ind[index]);
    }
}

File: cse107/EIUFING.py
def find_finger(random_number):
    finger = ["cai", "tro", "giua", "ap ut", "ut", "ut", "ap ut", "giua", "tro", "cai"]
    ind = ["trai", "phai"]

    finger_number = random_number % 18

    if 10 < finger_number <= 18:
        finger_number = 20 - finger_number
    index = 1 if finger_number > 5 else 0

    print("Ngon %s cua ban tay %s" % (finger[finger_number - 1], ind[index]))


if __name__ == "__main__":
    random_number = int(input())
    find_finger(random_number)
